pkg/api/api.go

```diff
@@ -3,115 +3,88 @@ package api
 import (
 	"context"
 	"fmt"
-	"log/slog"
 	"net/http"
 	"os"
 	"time"
 
-	"github.com/julienschmidt/httprouter"
+	"github.com/gorilla/mux"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/prometheus/common/version"
 	"github.com/re-cinq/cloud-carbon/pkg/config"
+	"github.com/re-cinq/cloud-carbon/pkg/log"
 )
 
+const readHeaderTimeout = 2 * time.Second
+
 // ApiServer object
 type API struct {
-	host        string
-	port        string
-	hostPort    string
+	*http.Server
+
+	addr        string
 	metricsPath string
-	server      *http.Server
 }
 
-// NewAPIServer instance
-func NewAPIServer() *API {
-	// Local variable
-	host := config.AppConfig().APIConfig.Address
-	port := config.AppConfig().APIConfig.Port
-
-	// Make sure the metrics path is set
-	metricsPath := config.AppConfig().APIConfig.MetricsPath
-	if metricsPath == "" {
-		metricsPath = "/metrics"
```

Comment:
Are you doing this with the viper set defaults?

Reply:
yes exactly!